Dockerfile

```diff
@@ -38,61 +38,57 @@ USER 0
+
+WORKDIR $REMOTE_SOURCES_DIR
+COPY $REMOTE_SOURCES $REMOTE_SOURCES_DIR
+COPY --from=deps $REMOTE_SOURCES_DIR $REMOTE_SOURCES_DIR
+RUN chmod +x $REMOTE_SOURCES_DIR/.yarn/releases/yarn-1.22.19.cjs
+RUN git config --global --add safe.directory $REMOTE_SOURCES_DIR
+RUN rm $REMOTE_SOURCES_DIR/app-config.yaml && mv $REMOTE_SOURCES_DIR/app-config.example.yaml $REMOTE_SOURCES_DIR/app-config.yaml
+
 RUN $YARN build --filter=backend
 
 # Stage 3 - Build the actual backend image and install production dependencies
 #@follow_tag(registry.redhat.io/ubi9/nodejs-18-minimal:1)
 FROM registry.access.redhat.com/ubi9/nodejs-18-minimal:1 AS runner
 USER 0
 
-# Env vars
-ENV YARN=./.yarn/releases/yarn-1.22.19.cjs
-ENV TECHDOCS_BUILDER_TYPE=external
-ENV TECHDOCS_GENERATOR_TYPE=local
-ENV TECHDOCS_PUBLISHER_TYPE=awsS3
-ENV SEGMENT_WRITE_KEY=temp
-ENV SEGMENT_TEST_MODE=true
-
 # Install gzip for tar and clean up
 RUN microdnf install -y gzip python3 python3-pip && \
     pip3 install mkdocs-techdocs-core==1.2.1 && \
     microdnf clean all
 
-COPY --from=build --chown=1001:1001 /opt/app-root/src/.yarn ./.yarn
-COPY --from=build --chown=1001:1001 /opt/app-root/src/.yarnrc.yml ./
+# Env vars
+ENV YARN=./.yarn/releases/yarn-1.22.19.cjs
+ENV REMOTE_SOURCES=.
+ENV REMOTE_SOURCES_DIR=/opt/app-root/src
 
-# Switch to nodejs user
-USER 1001
+WORKDIR $REMOTE_SOURCES_DIR
+COPY --from=build --chown=1001:1001 $REMOTE_SOURCES_DIR/.yarn $REMOTE_SOURCES_DIR/.yarn
+COPY --from=build --chown=1001:1001 $REMOTE_SOURCES_DIR/.yarnrc.yml $REMOTE_SOURCES_DIR/
+RUN chmod +x $REMOTE_SOURCES_DIR/.yarn/releases/yarn-1.22.19.cjs
 
 # Copy the install dependencies from the build stage and context
-COPY --from=build /opt/app-root/src/yarn.lock /opt/app-root/src/package.json /opt/app-root/src/packages/backend/dist/skeleton.tar.gz ./
+COPY --from=build --chown=1001:1001 $REMOTE_SOURCES_DIR/yarn.lock $REMOTE_SOURCES_DIR/package.json $REMOTE_SOURCES_DIR/packages/backend/dist/skeleton.tar.gz $REMOTE_SOURCES_DIR/
 RUN tar xzf skeleton.tar.gz && rm skeleton.tar.gz
 
-# Install production dependencies
-ENV IS_CONTAINER="TRUE"
-RUN $YARN install --frozen-lockfile --production --network-timeout 600000 --ignore-scripts && $YARN cache clean
-
 # Copy the built packages from the build stage
-COPY --from=build /opt/app-root/src/packages/backend/dist/bundle.tar.gz .
-RUN tar xzf bundle.tar.gz && rm bundle.tar.gz
+COPY --from=build --chown=1001:1001 $REMOTE_SOURCES_DIR/packages/backend/dist/bundle.tar.gz $REMOTE_SOURCES_DIR
+RUN tar xzf $REMOTE_SOURCES_DIR/bundle.tar.gz && rm $REMOTE_SOURCES_DIR/bundle.tar.gz
 
 # Copy any other files that we need at runtime
-COPY ./app-config.yaml ./app-config.production.yaml ./app-config.example.yaml ./app-config.example.production.yaml ./
+COPY --chown=1001:1001 $REMOTE_SOURCES/app-config.yaml $REMOTE_SOURCES/app-config.production.yaml $REMOTE_SOURCES/app-config.example.yaml $REMOTE_SOURCES/app-config.example.production.yaml $REMOTE_SOURCES_DIR
```

Comment:
do we need three versions of app-config in the final container?

Reply:
@nickboldt @schultzp2020 No the example files are not needed in the image. This is more to run the application locally.

Reply:
@christophe-f the example file is required for the helm chart.